code/modules/clothing/masks/miscellaneous.dm

```diff
@@ -18,7 +20,42 @@
 /obj/item/clothing/mask/muzzle/attack_hand(mob/user as mob)
 	if(user.wear_mask == src && !user.IsAdvancedToolUser())
 		return 0
+	else if(user.wear_mask == src && security_lock && locked)
+		if(ishuman(user))
+			if(doUnlock(user))
+				visible_message("<span class='danger'>[user] unlocks their [src.name].</span>", \
+									"<span class='userdanger'>[user] unlocks their [src.name].</span>")
+		else
+			to_chat(user, "You lack the ability to open the lock.")
+			return 0
 	..()
+	return 1
+
+/obj/item/clothing/mask/muzzle/proc/doUnlock(mob/living/carbon/human/user)
+	var/unlock = 0
+	var/obj/item/weapon/card/id/I = user.get_idcard(TRUE)
+	if(istype(I, /obj/item/weapon/card/emag))

```

Comment:
Won't this fail since you have `I` typecasted as a `obj/item/weapon/card/id` already?